src/files.rs

```diff
@@ -0,0 +1,1044 @@
+        sifis_quantized: f64,
+        crap: f64,
+        skunk: f64,
+        is_complex: bool,
+        coverage: f64,
+    ) -> Self {
+        Self {
+            sifis_plain,
+            sifis_quantized,
+            crap,
+            skunk,
+            is_complex,
+            coverage,
+        }
+    }
+}
+
+/// Struct with all the metrics computed for a single file
+#[derive(Clone, Default, Debug, Serialize, Deserialize)]
+#[allow(dead_code)]
+pub struct FileMetrics {
+    pub metrics: Metrics,
+    pub file: String,
+    pub file_path: String,
+}
+
+impl FileMetrics {
+    pub fn new(metrics: Metrics, file: String, file_path: String) -> Self {
+        Self {
+            metrics,
+            file,
+            file_path,
+        }
+    }
+
+    pub fn avg(m: Metrics) -> Self {
+        Self {
+            metrics: m,
+            file: "AVG".to_string(),
+            file_path: "-".to_string(),
+        }
+    }
+
+    pub fn min(m: Metrics) -> Self {
+        Self {
+            metrics: m,
+            file: "MIN".to_string(),
+            file_path: "-".to_string(),
+        }
+    }
+
+    pub fn max(m: Metrics) -> Self {
+        Self {
+            metrics: m,
+            file: "MAX".to_string(),
+            file_path: "-".to_string(),
+        }
+    }
+}
+
+type Output = (Vec<FileMetrics>, Vec<String>, Vec<FileMetrics>, f64);
+
+/// This Function get the folder of the repo to analyzed and the path to the json obtained using grcov
+/// It prints all the SIFIS, CRAP and SkunkScore values for all the files in the folders
+/// the output will be print as follows:
+/// FILE       | SIFIS PLAIN | SIFIS QUANTIZED | CRAP       | SKUNKSCORE | "IS_COMPLEX" | "PATH"
+/// if the a file is not found in the json that files will be skipped
+pub fn get_metrics_output(
+    metrics: Vec<FileMetrics>,
+    files_ignored: Vec<String>,
+    complex_files: Vec<FileMetrics>,
+) -> Result<()> {
+    println!(
+        "{0: <20} | {1: <20} | {2: <20} | {3: <20} | {4: <20} | {5: <20} | {6: <30}",
+        "FILE", "SIFIS PLAIN", "SIFIS QUANTIZED", "CRAP", "SKUNKSCORE", "IS_COMPLEX", "PATH"
+    );
+    metrics.iter().for_each(|m| {
+        println!(
+            "{0: <20} | {1: <20.3} | {2: <20.3} | {3: <20.3} | {4: <20.3} | {5: <20} | {6: <30}",
+            m.file,
+            m.metrics.sifis_plain,
+            m.metrics.sifis_quantized,
+            m.metrics.crap,
+            m.metrics.skunk,
+            m.metrics.is_complex,
+            m.file_path
+        );
+    });
+    println!("FILES IGNORED: {}", files_ignored.len());
+    println!("COMPLEX FILES: {}", complex_files.len());
+    Ok(())
+}
+
+/// This Function get the folder of the repo to analyzed and the path to the json obtained using grcov
+/// if the a file is not found in the json that files will be skipped
+/// It returns the  tuple (res, files_ignored, complex_files, project_coverage)
+pub fn get_metrics<A: AsRef<Path> + Copy, B: AsRef<Path> + Copy>(
```

Comment:
If we need to add a `Copy` trait better to pass a `PathBuf` because it means we have to save these paths

Reply:
Actually the Copy Trait was not needed so i removed it.